lib/inspec/profile.rb

```diff
@@ -89,7 +89,7 @@ def initialize(source_reader, options = {})
       @writable = options[:writable] || false
       @profile_id = options[:id]
       @cache = options[:cache] || Cache.new
-      @backend = options[:backend] || Inspec::Backend.create(options)
+      @backend = options[:backend] || Inspec::Backend.create(options.select { |k, _| k != 'target' })
```

Comment:
The fact that we hit (read: I created) this bug in the first place is because we blindly pass a hash of options that is created by the caller rather than curating a hash of options we actually need. Rather than deleting an option we don't need, I'd rather we only pass the options we **do** need. Otherwise, people like me will see calls to `#delete` without knowing why, there won't be a comment as to why, and then bugs are created. 🙂 

It's possible that we need to be actually creating the hash we need in `Inspec::Backend` rather than here.

@chris-rock / @arlimus - thoughts?